sw/airborne/firmwares/fixedwing/stabilization/stabilization_adaptive.c

```diff
@@ -520,3 +596,107 @@ inline static void h_ctl_pitch_loop(void)
   h_ctl_elevator_setpoint = TRIM_PPRZ(cmd);
 }
 
+/* yaw damper */
+#if H_CTL_YAW_LOOP
+inline static void h_ctl_yaw_loop(void)
+{
+ 
+#if H_CTL_YAW_TRIM_BY
+  // Actual Acceleration from IMU:
+ #ifndef SITL
+  struct Int32Vect3 accel_meas_body;
+  struct Int32RMat *body_to_imu_rmat = orientationGetRMat_i(&imu.body_to_imu);
+  int32_rmat_transp_vmult(&accel_meas_body, body_to_imu_rmat, &imu.accel);
+  float by = ACCEL_FLOAT_OF_BFP(accel_meas_body.y);
+ #else
+  float by = 0;
+ #endif
+
+  if (pprz_mode == PPRZ_MODE_MANUAL || launch == 0) {
+    h_ctl_yaw_by_sum_err = 0.;
+  } else {
+     if (h_ctl_yaw_by_igain > 0.){
+//     only update wehn: phi<60degrees and by<2g
+       if(fabsf(stateGetNedToBodyEulers_f()->phi) < 1.05 &&
+          fabsf(by) < 20.) {
+       h_ctl_yaw_by_sum_err += by * H_CTL_REF_DT;
+//     max half rudder deflection for trim
+       BoundAbs(h_ctl_yaw_by_sum_err, MAX_PPRZ/(2* h_ctl_yaw_by_igain));
+       }
+   } else {
+      h_ctl_yaw_by_sum_err = 0.;
+    }
+  }
+#endif
+
+#ifdef USE_AIRSPEED
+float Vo = *stateGetAirspeed_f();
+Bound(Vo, STALL_AIRSPEED, RACE_AIRSPEED);
+#else
+float Vo = NOMINAL_AIRSPEED;
+#endif
+
+  h_ctl_ref.yaw_rate = h_ctl_yaw_rate_setpoint // set by RC
+                       + 9.81f/ Vo * sinf(h_ctl_roll_setpoint) // for turns
+;
+  float d_err = h_ctl_ref.yaw_rate - stateGetBodyRates_f()->r;
+
+  float cmd = + h_ctl_yaw_dgain * d_err
+#if H_CTL_YAW_TRIM_BY
+              - h_ctl_yaw_by_igain * h_ctl_yaw_by_sum_err
+#endif
+;
+  cmd /= airspeed_ratio2;
+  h_ctl_rudder_setpoint = TRIM_PPRZ(cmd);
+}
+#endif
+
+/* CL with Flaps - direct lift control */
+#if H_CTL_CL_LOOP
+inline static void h_ctl_cl_loop(void)
+{
+
+#ifndef SITL
+  struct Int32Vect3 accel_meas_body;
+  struct Int32RMat *body_to_imu_rmat = orientationGetRMat_i(&imu.body_to_imu);
+  int32_rmat_transp_vmult(&accel_meas_body, body_to_imu_rmat, &imu.accel);
+  float bx = ACCEL_FLOAT_OF_BFP(accel_meas_body.z);
```

Comment:
the general idea was that for a 4 flap fixed wing Flaps could be moved by the Cl loop depending on the airspeed
if you are using it only in auto 1 you probably want to use the IAS to fly always with the best flap setting
if you plan to use auto 2 it is probably better to use the set point as this reduces actuator control movement and variation from the airspeed set point should be small 
in addition if you are flying circles with lets say 60 degrees you will need more lift (*sqrt g) at the same speed than in horizontal flight - that where bz is used to virtually reduce the IAS to get more Flap movement. I´ll correct the wrong bx its a typo
but as I stated, this is more an idea which I can't really test without a proper 4 flap plane.
I already asked Rolf if he could put this into his new glider to do a proper analysis if it really increases performance - and he will as soon as time permits.
its working fine on the bench though (tester with my ailerons set as flaps)